Add specialized directory asserts

valery1707 opened the pull request:
#### Check List:
* Fixes #1325
* Unit tests : YES
* Javadoc with a code example (API only) : YES

Method names are indicative - if there is a better/rather variants, then I'm ready to rename them.

Which version I should declare at API's JavaDoc?
Currently I declare `3.13.0`.

I plan to add (in a new PR) methods converting `FileAssert`/`PathAssert` to `IterableAssert<File>`/`IterableAssert<Path>` - what do you think about this?

valery1707 commented:
@joel-costigliola ping

joel-costigliola commented:
Thanks, I will review your PR shortly ! like by the end of the month likely

valery1707 commented:
@joel-costigliola ping

joel-costigliola commented:
@valery1707 sorry, I was too busy, hopefully I can have a look this weekend.
thanks for the reminder

joel-costigliola commented:
@valery1707 I had a look at the PR, I feel it provides too many assertions at the moment, I would prefer to narrow them down to the minimum, the rationale being that once added it is complicated to remove an assertion. 
A good strategy in that regard is adding the most useful assertions first and let users ask for missing ones.

I suggest we discuss what are the most useful assertions in #1325 to get feedback from users.

There is clearly some overlapping assertions in the PR:

- `isDirectoryContains(Predicate<Path> filter)`
- `isDirectoryContains(PathMatcher filter)`

I would probably remove the `PathMatcher` one as it looks like `isDirectoryContains(String syntaxAndPattern)` is covering it anyway.

We will also have to find a better name for ``isDirectoryContains` but let's discuss that in #1325.

Thanks for work!

joel-costigliola commented:
@valery1707 after the discussion in #1325, could you update your PR with the following changes:
- remove `isDirectoryContains(PathMatcher filter)`
- rename `isDirectoryContains` to `isDirectoryContaining`
- remove all negative assertions like `doesNotDirectoryContains`

Thanks!

valery1707 commented:
@joel-costigliola, done:
* Removed asserts with `FileFilter`, `FilenameFilter` and `PathMatcher`
* Renamed `isDirectoryContains` to `isDirectoryContaining`
* Renamed `doesNotDirectoryContains` to `isDirectoryNotContaining`
* Rebased onto current `master`
* Tag `@since` still have value `3.13.0`

valery1707 commented:
@joel-costigliola ping

joel-costigliola commented:
Integrated thanks @valery1707! (I improved some part of it FYI).

qwazer commented:
Waiting for release 3.13
